lib/SIL/SILPrinter.cpp

```diff
@@ -2891,29 +2905,37 @@ ID SILPrintContext::getID(const SILNode *node) {
     // Lazily initialize the instruction -> ID mapping.
     if (ValueToIDMap.empty())
       F->numberValues(ValueToIDMap);
-  } else {
-    setContext(BB);
-    // Lazily initialize the instruction -> ID mapping.
-    if (ValueToIDMap.empty()) {
-      unsigned idx = 0;
-      for (auto &I : *BB) {
-        // Give the instruction itself the next ID.
-        ValueToIDMap[&I] = idx;
-
-        // If there are no results, make sure we don't reuse that ID.
-        auto results = I.getResults();
-        if (results.empty()) {
-          idx++;
-
-        // Otherwise, assign all of the results an index.  Note that
-        // we'll assign the same ID to both the instruction and the
-        // first result.
-        } else {
-          for (auto result : results) {
-            ValueToIDMap[result] = idx++;
-          }
-        }
-      }
+    ID R = {ID::SSAValue, ValueToIDMap[node]};
+    return R;
+  }
+
+  setContext(BB);
+
+  // Check if we have initialized our ValueToIDMap yet. If we have, just use
+  // that.
+  if (!ValueToIDMap.empty()) {
+    ID R = {ID::SSAValue, ValueToIDMap[node]};
+    return R;
+  }
+
+  // Otherwise, initialize the instruction -> ID mapping cache.
+  unsigned idx = 0;
+  for (auto &I : *BB) {
+    // Give the instruction itself the next ID.
+    ValueToIDMap[&I] = idx;
+
+    // If there are no results, make sure we don't reuse that ID.
+    auto results = I.getResults();
+    if (results.empty()) {
+      idx++;
+      continue;
+    }
+
+    // Otherwise, assign all of the results an index.  Note that
+    // we'll assign the same ID to both the instruction and the
+    // first result.
+    for (auto result : results) {
+      ValueToIDMap[result] = idx++;
```

Comment:
Was this change just a de-indenting?  It looks fine to me, but I think it's probably unnecessarily confusing to do in the same patch as a bunch of multi-result compatibility fixes.

Reply:
I committed this separately. I just need to rebase.